packages/algolia/src/client.ts

```diff
@@ -21,29 +30,32 @@ export type EntityRecord<T extends keyof EntityResponses> =
 export type SearchEntityResponse<TEntityType extends keyof EntityResponses> =
   SearchResponse<EntityRecord<TEntityType>>;
 
-export const getEntityType = (
-  entity: EntityResponses[keyof EntityResponses],
-): keyof EntityResponses => {
-  if ('title' in entity && 'sharingStatus' in entity) {
-    return 'research-output';
-  }
-
-  return 'user';
+export type Payload<T extends EntityResponses[keyof EntityResponses]> = {
```

Comment:
yeah that wont work - i shouldnt be able to do this:
```typescript
const client = new AlgoliaSearchClient('asdas' as any);

client.save({ data: createResearchOutputResponse(), type: 'external-author' });
```

yet the definition for the `save` method allows it. If we need to be explicit we need to tie the given `type` to the `data`.

Reply:
updated